### ISSUE TITLE
wenet2.2.1 recognize.py uses attention decoding method, the result is abnormal

### ISSUE THREAD

**Issue by Yong2Fern** [wenet-e2e/wenet#2604]
Wenet2.2.1 version, the test set is librispeech's test_clean, the test results, the official pre-training model u2++conformer, two questions:
1. recognize.py uses attention mode to decode, and a large number of <sos/eos> symbols appear.
2. <sos/eos>=2 in the official pre-training model u2++conformer units.txt, and the decoding result using attention mode is empty.
answer：questions 2 is because <sos/eos>=2 in units.txt, when hyps goes to text_bpe, it goes to the break logic in recognize.py.
question1 is why ?????????

**Comment by xingchensong** [comment]
ctc 解码正常吗

**Comment by Yong2Fern** [comment]
官方下载的预训练声学模型，units.txt中<sos/eos>=2，recognize.py脚本中设置的结束符id是len(lang_char) - 1，ctc解码会出现尾部丢字的现象，修改recognize.py脚本结束符id=2，ctc_greedy_search/ctc_prefix_beam_search解码正常。attention解码完全异常。
